File: portal/portal/App_Code/DAO/SlideDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.IO;


namespace portal.App_Code.DAO
{
    public class SlideDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public Slide carregar(long pCodigo)
        {
            Slide obj = new Slide();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Slide where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Imagem = registros["imagem"].ToString();
                    obj.Titulo = registros["titulo"].ToString();
                    obj.Mensagem = registros["mensagem"].ToString();
                    obj.Url = registros["url"].ToString();
              
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<Slide> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<Slide> lista = new List<Slide>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Slide where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                    if (i != pFiltros.Length - 1)
                        sql = sql + " and ";
                    
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    Slide obj = new Slide();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Imagem = registros["imagem"].ToString();
                    obj.Titulo = registros["titulo"].ToString();
                    obj.Mensagem = registros["mensagem"].ToString();
                    obj.Url = registros["url"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(Slide obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into Slide(imagem, titulo, mensagem, url) value('{0}','{1}','{2}','{3}')";
                    sql = String.Format(sql, obj.Imagem, obj.Titulo, obj.Mensagem, obj.Url);
                }
                else
                {
                    sql = "update Slide set imagem='{0}', titulo='{1}', mensagem='{2}', url='{3}' where codigo={4}";
                    sql = String.Format(sql, obj.Imagem, obj.Titulo, obj.Mensagem, obj.Url, obj.Codigo);
                }

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(Slide obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from Slide where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }
}

File: portal/portal/App_Code/DAO/SlideDAO.php
<?php
namespace portal\App_Code\DAO {

//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
use portal\App_Code\Slide;
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);     

    try{
        require_once("/../Slide.php");        
        require_once("/../../connection.php");        
    }catch(Exception $e){
		require_once (dirname(__file__, 2) . "\Slide.php");
		require_once (dirname(__file__, 3) . "\connection.php");
    }

class SlideDAO{
//INSERIR--------------------------------------------------------------------------------------------------------------------------------------
function inserirSlide($slide){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into slide (imagem, titulo, mensagem, url, grupo, status) values (' " .
                 $slide->getImagem() . " ',' " .
                 $slide->getTitulo() . " ',' " .
                 $slide->getMensagem() . " ',' " .
                 $slide->getURL() . " ',' " .
                 $slide->getGrupo() . " ', " .
                 $slide->getTipoStatus() . " )";
                 mysqli_query($con,$sql);
                 echo "Inserção com sucesso!";
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);	
}

//ALTERAR---------------------------------------------------------------------------------------------------------------------------------------
    function alterarSlide($slide){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update slide set " . 
                "imagem = ' " . $slide->getImagem() . "', " .
                "grupo = ' " . $slide->getGrupo() . "', " .
                "titulo = ' " . $slide->getTitulo() . "', " .
                "mensagem = ' " . $slide->getMensagem() . "', " .
                "url = ' " . $slide->getURL() . "', " .
                "status =  " . $slide->getTipoStatus() . " " .
                "where codigo = " . $slide->getCodigo();
                mysqli_query($con,$sql);
                echo "Dados alterados com sucesso";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }    
    mysqli_close($con);	
    }

//CARREGAR--------------------------------------------------------------------------------------------------------------------------------------
    function carregarSlide ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from slide where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                $slider = new Slide();
                    if($carga = mysqli_fetch_array($gargantula)){
                        $slider->setSlide(($carga["codigo"]),
                                          ($carga["imagem"]),
                                          ($carga["grupo"]),
                                          ($carga["titulo"]),
                                          ($carga["mensagem"]),
                                          ($carga["url"]),
                                          ($carga["status"]));
                        mysqli_close($con);
                        return $slider;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;            
            }
mysqli_close($con);
    }

//CARREGAR LISTA---------------------------------------------------------------------------------------------------------------------------------
    function carregarListaSlide($qtdOpcoes, $ordem){
        $sql = "select * from slide where ";
        
        for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++ ){
            $sql = $sql . $qtdOpcoes[$i];
			     if($i != $qtdOpcoes[$i] - 1){
                    $sql = $sql . "and ";  
                 } 
        }
        
        $sql = $sql . " order by " . $ordem;
        
        try{
            $con = BD_AbrirConexao();
            $gargantula = mysqli_query($con,$sql);
                if($carga = mysqli_fetch_array($gargantula)){
                   $listaSlide = [];
                        while($carga = mysqli_fetch_array($gargantula)){
                              $slider = new Slide();
                              $slider->setSlide(($carga["codigo"]),
                                                ($carga["imagem"]),
                                                ($carga["grupo"]),
                                                ($carga["titulo"]),
                                                ($carga["mensagem"]),
                                                ($carga["url"]),
                                                ($carga["status"]));
                              $listaSlide[] = $slider;
                        }
                    mysqli_close($con); 
                    return $listaSlide;
                }
        }catch(Exception $e){
                echo "Erro ao carregar dados: " . $e;
            }
        mysqli_close($con);        
    }

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarSlide($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from slide where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
                echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;
            }
        mysqli_close($con);
    }
  }
}
?>